### PR TITLE
Inprove the error message for GPDB's partition syntax

### PR DESCRIPTION
**Pull request by JunfengYang**
Remove error message related GPDB_12_MERGE_FIXME.
One of the case the error check is duplicated. So just remove it.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/parser/parse_partition_gp.c [hunk 1]
```diff
@@ -993,7 +993,7 @@ generateRangePartitions(ParseState *pstate,
 	if (partnamecomp->tablename == NULL && boundspec->partEvery)
 	{
 		if (list_length(boundspec->partEvery) != partkey->partnatts)
-			elog(ERROR, "invalid number of every values"); // GPDB_12_MERGE_FIXME: improve message
+			elog(ERROR, "number of EVERY values should cover all partition key columns");
```
Comment (ashwinstar) [inline src/backend/parser/parse_partition_gp.c:996]: ereport?

File: src/backend/commands/tablecmds_gp.c [hunk 2]
```diff
@@ -869,8 +873,6 @@ AtExecGPSplitPartition(Relation rel, AlterTableCmd *cmd)
 						convert_exclusive_start_inclusive_end(startConst,
 															  part_col_typid, part_col_typmod,
 															  true);
-					if (startConst->constisnull)
```
Comment (ashwinstar) [inline src/backend/commands/tablecmds_gp.c:872]: So, we will never have this condition? Do we need to assert for it instead then.

File: src/backend/commands/tablecmds_gp.c [hunk 3]
```diff
@@ -809,7 +809,9 @@ AtExecGPSplitPartition(Relation rel, AlterTableCmd *cmd)
 					Const *endConst;
 
 					if (list_length(end) != partkey->partnatts)
-						elog(ERROR, "invalid number of end values"); // GPDB_12_MERGE_FIXME: improve message
+						ereport(ERROR,
```
Comment (ashwinstar) [inline src/backend/commands/tablecmds_gp.c:812]: Is there a way for us to print `parser_errposition()` for these?

File: src/backend/commands/tablecmds_gp.c (hunk 2 above)
Comment (JunfengYang) [reply src/backend/commands/tablecmds_gp.c:872]: Line 866 already check this condition.

File: src/backend/commands/tablecmds_gp.c (hunk 3 above)
Comment (JunfengYang) [reply src/backend/commands/tablecmds_gp.c:812]: Let me try.

Comment (JunfengYang) [reply src/backend/commands/tablecmds_gp.c:812]: For the `AlterTable` which called under `ProcessUtilitySlow`, the `queryString` is not passed in.
So, our code can not set `pstate->p_sourcetext` with the `queryString`, and when calling `parser_errposition`, it'll return directly.

```
parser_errposition


/* Can't do anything if source text is not available */
	if (pstate == NULL || pstate->p_sourcetext == NULL)
		return;
```
All the `parser_errposition()` under tablecmds_gp.c is actually useless. Unless we modify the `AlterTable` which will cause diffs with upstrem. Any opinions?
I'll add `parse_errposition` for the `ereport` for now.

Comment (JunfengYang) [reply src/backend/commands/tablecmds_gp.c:814]: Found another solution to just pass `queryString` through `transformAlterTableStmt`.

**Review by adam8157 — APPROVED** [review]

**Pull request merged**